typing/ctype.ml

```diff
@@ -3261,23 +3263,48 @@ let expand_head_trace env t =
    (2) the original label is not optional
 *)
 
-let filter_arrow env t l =
+let filter_arrow env t l ~force_poly =
```

Comment:
There is no such precondition: instead we expect the caller to fix up the invariant after. I've added some comments to make this clearer.

Reply:
Hmm... I don't see these comments on the branch (nor in the PR). Maybe you forgot to push? I'll stop making responses in case we're somehow out of sync.

Reply:
The comments are in the mli file and at the call site where `force_poly` might be false.